mmeds/spawn.py

```diff
@@ -164,11 +164,16 @@ def log_processes(self):
         # There is a seperate log of processes for each day
         current_log = fig.PROCESS_LOG_DIR / (datetime.now().strftime('%Y%m%d') + '.yaml')
         if current_log.exists():
-            with open(current_log, 'r') as f:
-                finished += yaml.safe_load(f)
+            try:
+                finished += yaml.safe_load(current_log.read_text())
+            # TODO Figure out why this is happening
+            except TypeError:
+                logger.error('Error loading process log {}'.format(current_log))
 
-        with open(current_log, 'w+') as f:
-            yaml.dump(finished, f)
+        # Only create the file if there are processes to log
+        if finished:
+            with open(current_log, 'w+') as f:
+                yaml.dump(finished, f)
```

Comment:
Not sure if you have encountered this, but yaml.dump() by default won't indent 4 spaces and will reorder the dictionary keys at the highest level.

Reply:
although if you're not working with configuration files directly, may not be an issue.

Reply:
That is indeed not an issue for me at this time. I have some plans to expand config file functionality going forward but not in this PR